python/lsst/atmospec/processStar.py

```diff
@@ -711,15 +717,46 @@ def getNormalizedTargetName(self, target):
             return converted
         return target
 
+    def _getSpectractorTargetSetting(self, inputCentroid):
+        """Calculate the value to set SPECTRACTOR_FIT_TARGET_CENTROID to.
+
+        Parameters
+        ----------
+        inputCentroid : `???`
+            The input centroid to the task.
+
+        Returns
+        -------
+        centroidMethod : `str`
+            The value to set SPECTRACTOR_FIT_TARGET_CENTROID to.
+        """
+
+        # if mode is auto and the astrometry worked then it's an exact
+        # centroid, and otherwise we fit, as per docs on this option.
+        if self.config.targetCentroidMethod == 'auto':
+            if inputCentroid['astrometricMatch'] is True:
+                self.log.info("Auto centroid is using exact centroid for target from the astrometry")
+                return 'guess'  # this means exact
+            else:
+                self.log.info("Auto centroid is using FIT in Spectractor to get the target centroid")
+                return 'fit'  # this means exact
+
+        # this is just renaming the config parameter because guess sounds like
```

Comment:
This is confusing, but from this comment and those above, it seems that is just how it is.